### PR TITLE
Add request headers to logger

### PR DESCRIPTION
**Pull request by ekampp**
This will adds headers to the log output. 

I don't believe I have the Rails headers included in this. Any suggestions on how to add these? 

NB: Current, failing specs where failing before I started my changes.

### CODE REVIEW

File: spec/integration/lib/grape/middleware/logger_spec.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@
     expect(subject.logger).to receive(:info).with ''
     expect(subject.logger).to receive(:info).with %Q(Started POST "/api/1.0/users" at #{subject.start_time})
     expect(subject.logger).to receive(:info).with %Q(Processing by TestAPI/users)
+    expect(subject.logger).to receive(:info).with %Q(     Headers: {})
```
Comment (ridiculous) [inline spec/integration/lib/grape/middleware/logger_spec.rb]: It'd be nice to test this with actual data, to show it working

File: lib/grape/middleware/logger.rb [hunk 2]
```diff
@@ -80,6 +81,10 @@ def after_failure(error)
     after(error[:status])
   end
 
+  def headers
+    env['grape.request.headers']
```
Comment (ridiculous) [inline lib/grape/middleware/logger.rb]: What do these headers look like? Could it be potentially very large?

Comment (ekampp) [inline lib/grape/middleware/logger.rb]: Yes, if the request contains a ton of headers then there will be a ton of data here. It also currently only supports Grape. 

Any preferences on the specific headers you'd like here? Perhaps a configuration option with sensible defaults.

**Comment by ekampp** [comment]
I'm perfectly fine with opt-in. I made the branch for my self, so just made the PR as a sort of "thous could be useful for others, perhaps?"

File: spec/integration/lib/grape/middleware/logger_spec.rb (hunk 1 above)
Comment (ekampp) [inline spec/integration/lib/grape/middleware/logger_spec.rb]: I had some trouble tracing down this bit to begin with. Any pointers on where to spoof some testing headers?

File: lib/grape/middleware/logger.rb (hunk 2 above)
Comment (ridiculous) [inline lib/grape/middleware/logger.rb]: If we make this opt in, it shouldn't really matter. Just wondering how noisy this makes the log

File: spec/integration/lib/grape/middleware/logger_spec.rb (hunk 1 above)
Comment (ridiculous) [inline spec/integration/lib/grape/middleware/logger_spec.rb]: Yeah, checkout the factory here: https://github.com/ridiculous/grape-middleware-logger/blob/master/spec/factories.rb#L31

File: lib/grape/middleware/logger.rb (hunk 2 above)
Comment (ridiculous) [inline lib/grape/middleware/logger.rb]: Although, we should use the grape constants for the hash key (see `#parameters`)

**Comment by ridiculous** [comment]
Cool, yeah, let's do this as opt-in. Please document the new option in the README.

What's the failing spec you mentioned in the description?

**Comment by ridiculous** [comment]
Oh, that reminds me, I need to add a changelog file for this project. People have been requesting it and I think it'd be useful. If you're up to it, that'd be a great addition as well 😁

**Comment by ekampp** [comment]
I don't remember the failing spec in particular, but there where two of them on my local system. Clearly they're not there on the CI.

**Comment by ridiculous** [comment]
Any updates on this? I've since added a changelog, so no need for that. But would be cool to get this in as an optional config

**Comment by ekampp** [comment]
No update yet. I will be implementing this, but I haven't had time recently. Sorry.

**Comment by nbulaj** [comment]
I think it must be configurable option (include headers or not in the log). `Rails`, for example, doesn't log headers by default.

**Comment by yamamotok** [comment]
I came to need request header logging and tried to improve the idea of this pull request with configurable option. It is still in my own branch but, if it's OK, I will send a pull request.
https://github.com/visits-works/grape-middleware-logger/commits/feature/with_header_option

**Comment by ridiculous** [comment]
Hey @yamamotok, it looks really good! I like it. Just 2 pieces of feedback:

1. How about calling the option `:headers` vs `:with_headers`
1. Let's change`"all"` to `:all`, because a symbol is more appropriate for special values
1. Let's use coercion instead of type checking (`Array(@options[:headers])` vs `@options[:headers].is_a?(Array)`

If we can make those changes, I'd be happy to merge it :smile:

**Comment by yamamotok** [comment]
Thank you all! 
After making some changes, I sent a pull request; https://github.com/ridiculous/grape-middleware-logger/pull/17
Can you take a look in your spare time?

**Comment by ekampp** [comment]
@yamamotok remember to add a closure note in your PR pointing to this one, so this will be closed when your PR is closed. Thanks 👍

**Comment by yamamotok** [comment]
Sure, I will update it later. Thank you @ekampp !

**Comment by ridiculous** [comment]
Implemented in #17

**Pull request closed without merging**